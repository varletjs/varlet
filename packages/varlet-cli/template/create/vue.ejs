<template>
  <div class="var-<%- kebabCaseName %>"></div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import './<%- kebabCaseName %>.less'

export default defineComponent({
  name: '<%- namespace + bigCamelizeName %>',
})
</script>
