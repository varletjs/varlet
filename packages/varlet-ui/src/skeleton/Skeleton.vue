<template>
  <div class="var--box var-skeleton">
    <div class="var-skeleton__data" v-if="!loading">
      <slot />
    </div>

    <div class="var-skeleton__content" v-if="loading && !fullscreen">
      <div class="var-skeleton__card" :style="{ height: toSizeUnit(cardHeight) }" v-if="card">
        <div class="var-skeleton--animation"></div>
      </div>
      <div class="var-skeleton__article">
        <div
          class="var-skeleton__avatar"
          :style="{
            width: toSizeUnit(avatarSize),
            height: toSizeUnit(avatarSize),
          }"
          v-if="avatar"
        >
          <div class="var-skeleton--animation"></div>
        </div>
        <div class="var-skeleton__section">
          <div class="var-skeleton__title" :style="{ width: toSizeUnit(titleWidth) }" v-if="title">
            <div class="var-skeleton--animation"></div>
          </div>
          <div
            class="var-skeleton__row"
            v-for="(r, index) in toNumber(rows)"
            :key="r"
            :style="{ width: toSizeUnit(rowsWidth[index]) }"
          >
            <div class="var-skeleton--animation"></div>
          </div>
        </div>
      </div>
    </div>

    <div class="var-skeleton__fullscreen" :style="{ zIndex: toNumber(fullscreenZIndex) }" v-if="loading && fullscreen">
      <div class="var-skeleton--animation"></div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { props } from './props'
import { toSizeUnit } from '../utils/elements'
import { toNumber } from '../utils/shared'

export default defineComponent({
  name: 'VarSkeleton',
  props,
  setup() {
    return {
      toSizeUnit,
      toNumber,
    }
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import './skeleton';
</style>
