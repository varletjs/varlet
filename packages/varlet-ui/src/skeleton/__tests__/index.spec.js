import example from '../example'
import Skeleton from '..'
import VarSkeleton from '../Skeleton'
import { mount } from '@vue/test-utils'
import { createApp, h } from 'vue'

test('test skeleton example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test skeleton plugin', () => {
  const app = createApp({}).use(Skeleton)
  expect(app.component(Skeleton.name)).toBeTruthy()
})

test('test skeleton loading', async () => {
  const wrapper = mount(VarSkeleton, {
    props: {
      loading: true,
    },
    slots: {
      default: () => h('div', { class: 'skeleton-default-slot' }),
    },
  })

  expect(wrapper.find('.skeleton-default-slot').exists()).toBeFalsy()

  await wrapper.setProps({ loading: false })
  expect(wrapper.find('.skeleton-default-slot').exists()).toBeTruthy()

  wrapper.unmount()
})

test('test skeleton styles', () => {
  const wrapper = mount(VarSkeleton, {
    props: {
      title: true,
      card: true,
      avatar: true,
      titleWidth: '100px',
      cardHeight: '100px',
      avatarSize: '100px',
      rows: 2,
      rowsWidth: ['600px', '500px'],
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test skeleton fullscreen', () => {
  const wrapper = mount(VarSkeleton, {
    props: {
      fullscreen: true,
      fullscreenZIndex: 1000,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})
