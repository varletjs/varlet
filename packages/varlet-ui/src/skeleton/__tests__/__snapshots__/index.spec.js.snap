// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test skeleton example 1`] = `
"<div class=\\"app-type\\">基本使用</div>
<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <!--v-if-->
    <div class=\\"var-skeleton__article\\">
      <!--v-if-->
      <div class=\\"var-skeleton__section\\">
        <!--v-if-->
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">显示标题</div>
<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <!--v-if-->
    <div class=\\"var-skeleton__article\\">
      <!--v-if-->
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">自定义段落高度</div>
<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <!--v-if-->
    <div class=\\"var-skeleton__article\\">
      <!--v-if-->
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 200px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 100px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 50px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">显示头像</div>
<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <!--v-if-->
    <div class=\\"var-skeleton__article\\">
      <div class=\\"var-skeleton__avatar\\">
        <div class=\\"var-skeleton--animation\\"></div>
      </div>
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">显示卡片</div>
<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <div class=\\"var-skeleton__card\\">
      <div class=\\"var-skeleton--animation\\"></div>
    </div>
    <div class=\\"var-skeleton__article\\">
      <div class=\\"var-skeleton__avatar\\">
        <div class=\\"var-skeleton--animation\\"></div>
      </div>
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">全屏模式</div><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--danger var-elevation--2 button\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">切换全屏模式</div>
</button>
<div class=\\"var--box var-skeleton\\">
  <div class=\\"var-skeleton__data\\"></div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`test skeleton fullscreen 1`] = `
"<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"var-skeleton__fullscreen\\" style=\\"z-index: 1000;\\">
    <div class=\\"var-skeleton--animation\\"></div>
  </div>
</div>"
`;

exports[`test skeleton styles 1`] = `
"<div class=\\"var--box var-skeleton\\">
  <!--v-if-->
  <div class=\\"var-skeleton__content\\">
    <div class=\\"var-skeleton__card\\" style=\\"height: 100px;\\">
      <div class=\\"var-skeleton--animation\\"></div>
    </div>
    <div class=\\"var-skeleton__article\\">
      <div class=\\"var-skeleton__avatar\\" style=\\"width: 100px; height: 100px;\\">
        <div class=\\"var-skeleton--animation\\"></div>
      </div>
      <div class=\\"var-skeleton__section\\">
        <div class=\\"var-skeleton__title\\" style=\\"width: 100px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 600px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
        <div class=\\"var-skeleton__row\\" style=\\"width: 500px;\\">
          <div class=\\"var-skeleton--animation\\"></div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
